library/ssl_tls13_generic.c

```diff
@@ -1398,11 +1398,11 @@ int mbedtls_ssl_reset_transcript_for_hrr( mbedtls_ssl_context *ssl )
 #else
         mbedtls_sha256_starts( &ssl->handshake->fin_sha256, 0 );
 #endif
-#endif /* MBEDTLS_HAS_ALG_SHA_256_VIA_MD_OR_PSA_BASED_ON_USE_PSA */
     }
+#endif /* MBEDTLS_HAS_ALG_SHA_256_VIA_MD_OR_PSA_BASED_ON_USE_PSA */
+#if defined(MBEDTLS_HAS_ALG_SHA_384_VIA_MD_OR_PSA_BASED_ON_USE_PSA)
```

Comment:
I think the `else` needs to be before the `#endif #if` sequence, or we're going to have compilation errors in configs without SHA-256.

Reply:
Indeed this else should have cause failure if `!MBEDTLS_HAS_ALG_SHA_256_VIA_MD_OR_PSA_BASED_ON_USE_PSA`.

But if we move it from from 1404 and put it after 1401 then when `!MBEDTLS_HAS_ALG_SHA_384_VIA_MD_OR_PSA_BASED_ON_USE_PSA` this else will be related to line 1418.

Maybe it can be simply removed? We are choosing operation based on `ciphersuite_info->mac` so even if both are supported only one is selected.

Reply:
Yeah, I agree we can just remove the `else`, just the `if`s are enough, and that way things are more regular.